lib/public/DB/QueryBuilder/Events/BeforeQueryExecuted.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Andrew Summers
+ *
+ * @author Andrew Summers
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCP\DB\QueryBuilder\Events;
+
+use OCP\EventDispatcher\Event;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use Doctrine\DBAL\Result;
+
+/**
+ * This event is used by apps to intercept, inspect, and potentially modify
+ * database queries prior to execution. This can be used for deep integrations,
+ * restricting user access to certain data, redacting information, etc.
+ *
+ * The result field can optionally be set by the event listener by executing the
+ * query in the event handler logic. This allows apps to inspect the results,
+ * use try/catch blocks around the query execution, and/or modify/re-execute the
+ * query if necessary. If the result field is not set, the QueryBuilder class
+ * will execute the query as expected by default.
+ *
+ * @see https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/projects.html
+ * @since 27.0.0
+ */
+class BeforeQueryExecuted extends Event {
```

Comment:
According to https://docs.nextcloud.com/server/latest/developer_manual/basics/events.html#naming-scheme this should be named:
```suggestion
class BeforeQueryExecutedEvent extends Event {
```

Reply:
> While I agree, I think the idea here was to exactly do that. The planned app rewrites the queries and decrypts the results where needed

This is correct. However, I CAN get by with just executing the query in my app and setting the result - the QueryBuilder::execute method checks for the existence of a result and uses that if necessary. I added the setter just as a convenience. I believe I can take it out and still have the same functionality.